t/api/02-iso.t

```diff
@@ -724,4 +725,55 @@ subtest 'Catch blocked_by cycles' => sub {
     $schema->txn_rollback;
 };
 
+subtest 'Create dependency for jobs on different machines' => sub {
+    $schema->txn_begin;
+    $t->post_ok('/api/v1/machines', form => {name => '64bit-ipmi', backend => 'ipmi', 'settings[TEST]' => 'ipmi'})
+      ->status_is(200);
+    add_opensuse_test('supportserver1');
+    add_opensuse_test('supportserver2', MACHINE => '64bit-ipmi');
+    add_opensuse_test('client', PARALLEL_WITH => 'supportserver1,supportserver2', MACHINE => 'Laptop_64');
+
+    add_opensuse_test('test1');
+    add_opensuse_test('test2', MACHINE          => '64bit-ipmi');
+    add_opensuse_test('test3', START_AFTER_TEST => 'test1,test2');
+
+    my $res = schedule_iso(
+        {
+            ISO     => $iso,
+            DISTRI  => 'opensuse',
+            VERSION => '13.1',
+            FLAVOR  => 'DVD',
+            ARCH    => 'i586',
+            BUILD   => '0091',
+            _GROUP  => 'opensuse test',
+        });
+
+    is($res->json->{count}, 6, '6 jobs scheduled');
+    my @newids = @{$res->json->{ids}};
+    my $newid  = $newids[0];
+
+    $ret = $t->get_ok('/api/v1/jobs');
+    my @jobs = @{$ret->tx->res->json->{jobs}};
+
+    my $server_64     = find_job(\@jobs, \@newids, 'supportserver1', '64bit');
+    my $server_ipmi   = find_job(\@jobs, \@newids, 'supportserver2', '64bit-ipmi');
+    my $client_laptop = find_job(\@jobs, \@newids, 'client',         'Laptop_64');
+    is_deeply(
+        $client_laptop->{parents},
+        {Parallel => [$server_64->{id}, $server_ipmi->{id}], Chained => []},
+        "server_64 and server_ipmi are the parents of client_laptop"
+    );
+
+    my $test1_64   = find_job(\@jobs, \@newids, 'test1', '64bit');
+    my $test2_ipmi = find_job(\@jobs, \@newids, 'test2', '64bit-ipmi');
+    my $test3_64   = find_job(\@jobs, \@newids, 'test3', '64bit');
+    is_deeply(
+        $test3_64->{parents},
+        {Parallel => [], Chained => [$test1_64->{id}, $test2_ipmi->{id}]},
```

Comment:
the test looks pretty promising, but I'm just as concerned about the error cases.

e.g. a job group with install_ltp@powerpc and use_ltp@powerpc and use_ltp@64bit. This should create a starts_after dependency for both use_ltps to the one install_ltp

But if there are 2 install_ltp and 2 use_ltp, the best matches should be taken. I.e. if two are on the same machine, those should be taken. If 2 are left, the machine shall be ignored. If 4 jobs are on 4 different machines, we should error as we can't find the dots.